### PR TITLE
Tests for GoodLoggerCheck ChecksStyle Rule

### PR DESCRIPTION
**Pull request by mssfang**
Add tests for GoodLoggerCheck ChecktStyle Rule.

### CODE REVIEW

File: eng/code-quality-reports/src/test/resources/com/azure/tools/checkstyle/checks/GoodLoggerCheck/ExternalLoggerLibrary.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import com.azure.core.util.logging.ClientLogger;
+// invalid external logger library
+import org.apache.logging.log4j;
+import org.slf4j;
+import java.util.logging;
+
```
Comment (JonathanGiles) [inline eng/code-quality-reports/src/test/resources/com/azure/tools/checkstyle/checks/GoodLoggerCheck/ExternalLoggerLibrary.java]: You should at least make this a valid class file.

File: eng/code-quality-reports/src/test/resources/com/azure/tools/checkstyle/checks/GoodLoggerCheck/WrongClassInLoggerConstructor.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import com.azure.core.util.logging.ClientLogger;
+
+public class WrongClassInLoggerConstructor {
+    // wrong class in ClientLogger constructor
+    private final ClientLogger logger = new ClientLogger(XXXXXX.class);
```
Comment (JonathanGiles) [inline eng/code-quality-reports/src/test/resources/com/azure/tools/checkstyle/checks/GoodLoggerCheck/WrongClassInLoggerConstructor.java]: Does this class actually compile - XXXXX.class presumably doesn't exist?

Comment (mssfang) [reply eng/code-quality-reports/src/test/resources/com/azure/tools/checkstyle/checks/GoodLoggerCheck/WrongClassInLoggerConstructor.java]: These java files in the resource folder don't need to be compiled. It is treated as a text file. CheckStyle just parse the text to be Abstract Syntax Tree and valid with the rules.

File: eng/code-quality-reports/src/test/java/com/azure/tools/checkstyle/checks/GoodLoggerCheckTests.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Test
+    public void invalidExternalLoggers() throws Exception {
+        String[] expected = {
+            expectedErrorMessage(3,1, String.format(EXTERNAL_LOGGER_USED_MESSAGE,
+                "org.apache.logging.log4j")),
+            expectedErrorMessage(4,1, String.format(EXTERNAL_LOGGER_USED_MESSAGE, "org.slf4j")),
+            expectedErrorMessage(5,1, String.format(EXTERNAL_LOGGER_USED_MESSAGE, "java.util.logging"))
+        };
+        verify(checker, getPath("ExternalLoggerLibrary.java"), expected);
```
[the remark this answers is not in the document]
Comment (mssfang) [reply eng/code-quality-reports/src/test/java/com/azure/tools/checkstyle/checks/GoodLoggerCheckTests.java]: Will do. Thanks

File: eng/code-quality-reports/src/test/resources/com/azure/tools/checkstyle/checks/GoodLoggerCheck/ExternalLoggerLibrary.java (hunk 1 above)
Comment (mssfang) [reply eng/code-quality-reports/src/test/resources/com/azure/tools/checkstyle/checks/GoodLoggerCheck/ExternalLoggerLibrary.java]: Trying to without the valid java class file to remain self this is not the java file but a text file that fake the java file to create the Abstract Syntax Tree.
But I will add the class to make a valid java class.

File: eng/code-quality-reports/src/test/java/com/azure/tools/checkstyle/checks/GoodLoggerCheckTests.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
+package com.azure.tools.checkstyle.checks;
```
Comment (conniey) [inline eng/code-quality-reports/src/test/java/com/azure/tools/checkstyle/checks/GoodLoggerCheckTests.java]: You're missing that MIT header on all your files.

Comment (mssfang) [reply eng/code-quality-reports/src/test/java/com/azure/tools/checkstyle/checks/GoodLoggerCheckTests.java]: We are only check GoodLoggerCheck. This test data will not follow other rules but only point out the GoodLoggerCheck. It is a fake java file.

Comment (mssfang) [reply eng/code-quality-reports/src/test/java/com/azure/tools/checkstyle/checks/GoodLoggerCheckTests.java]: These faked java files are under resource folder will be not be complied.

File: eng/code-quality-reports/src/test/resources/com/azure/tools/checkstyle/checks/GoodLoggerCheck/InvalidLoggerNameTestData.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import com.azure.core.util.logging.ClientLogger;
+
+public class InvalidLoggerNameTestData {
+    // invalid LOGGER name
+    private final ClientLogger LOGGER = new ClientLogger(InvalidLoggerNameTestData.class);
```
Comment (alzimmermsft) [inline eng/code-quality-reports/src/test/resources/com/azure/tools/checkstyle/checks/GoodLoggerCheck/InvalidLoggerNameTestData.java]: Do we need to check this one as the built-in Checkstyle rule will throw an error here about incorrect casing

File: eng/code-quality-reports/src/test/java/com/azure/tools/checkstyle/checks/GoodLoggerCheckTests.java (hunk 4 above)
Comment (alzimmermsft) [reply eng/code-quality-reports/src/test/java/com/azure/tools/checkstyle/checks/GoodLoggerCheckTests.java]: This is the test class and should be checked for the license header

File: eng/code-quality-reports/src/test/resources/com/azure/tools/checkstyle/checks/GoodLoggerCheck/InvalidLoggerNameTestData.java (hunk 5 above)
Comment (mssfang) [reply eng/code-quality-reports/src/test/resources/com/azure/tools/checkstyle/checks/GoodLoggerCheck/InvalidLoggerNameTestData.java]: Thanks. Will change the var name to others, like "abcdLogger"

File: eng/code-quality-reports/src/test/java/com/azure/tools/checkstyle/checks/GoodLoggerCheckTests.java (hunk 4 above)
Comment (mssfang) [reply eng/code-quality-reports/src/test/java/com/azure/tools/checkstyle/checks/GoodLoggerCheckTests.java]: My bad. This is not the fake java files. Interesting. The test still passed CI

**Pull request merged**